agent/src/control_interface/pipes_channel_task.rs

```diff
@@ -74,11 +80,23 @@ impl PipesChannelTask {
                     if let Ok(to_ankaios) = decode_to_server(to_ankaios_binary) {
                         match to_ankaios.try_into() {
                             Ok(ToAnkaios::Request(mut request)) => {
-                                request.prefix_request_id(&self.request_id_prefix);
-                                let _ = self.output_pipe_channel.send(ToServer::Request(request)).await;
+                                if self.authorizer.authorize(&request) {
+                                    log::debug!("Allowing request '{:?}' from authorizer '{:?}'", request, self.authorizer);
+                                    request.prefix_request_id(&self.request_id_prefix);
+                                    let _ = self.output_pipe_channel.send(ToServer::Request(request)).await;
+                                } else {
+                                    log::debug!("Denying request '{:?}' from authorizer '{:?}'", request, self.authorizer);
+                                    let error = Response {
+                                        request_id: request.request_id,
+                                        response_content: ResponseContent::Error(Error {
+                                            message: "Access denied".into(),
+                                        }),
+                                    };
+                                    let _ = self.forward_from_server(error).await;
+                                };
```

Comment:
I cannot find any specific unit tests to test the accept/deny logic.

Reply:
Will also be done in a separate PR.